core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementService.java

```diff
@@ -0,0 +1,117 @@
+package com.sequenceiq.cloudbreak.service.publicendpoint;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.loadbalancer.LoadBalancer;
+import com.sequenceiq.cloudbreak.service.environment.EnvironmentClientService;
+import com.sequenceiq.cloudbreak.service.stack.LoadBalancerPersistenceService;
+import com.sequenceiq.common.api.type.LoadBalancerType;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.freeipa.api.v1.dns.DnsV1Endpoint;
+import com.sequenceiq.freeipa.api.v1.dns.model.AddDnsARecordRequest;
+import com.sequenceiq.freeipa.api.v1.dns.model.AddDnsCnameRecordRequest;
+
+@Service
+public class FreeIPAEndpointManagementService extends BasePublicEndpointManagementService {
```

Comment:
`DnsV1Endpoint` expects environment edit right from the user right now. If a datahub admin or similar can launch cluster who doesn't have env edit right I think these calls would fail.
I think a good solution would be to make these calls as internal actor.
cc @horadla23 @keyki @jamisonbennett

Reply:
precisely, please use `ThreadBasedUserCrnProvider.doAsInternActor(() -> method)`

Reply:
I've put these wrappers in `ClusterPublicEndpointManagementService`, where it's calling this service. For example:
````
ThreadBasedUserCrnProvider.doAsInternalActor(() ->
  freeIPAEndpointManagementService.deleteLoadBalancerDomainFromFreeIPA(stack));
````

Reply:
I'm not sure this is a good idea to change those instead of the calls which are really meant to be internal. What do you think @horadla23 ?

Reply:
in long term i have to agree with @lacikaaa . if someone changes service logic and wants to get the initiator user crn for example, then it will return with false result. the best is to change only the endpoint calls to internal

Reply:
also please check if DnsV1Controller is prepared for internal calls based on the readme I provided (meaning that the proper parameter is annotated to be able to get current account id instead of static account id of internal actor). For me it seems it is not, and the DnsV1Endpoint call will fail some accountid issue

Reply:
I will do the freeipa part in `DnsV1Controller` to work with internal actor asap

Reply:
on PR: https://github.com/hortonworks/cloudbreak/pull/9740

Reply:
@lacikaaa @horadla23 I've read the linked document. I'm afraid I couldn't quite follow it though, and am not sure what I should be doing.

The registration method is only called by `StackV4Controller`, and the delete method looks like it could be triggered by either `StackV4Controller` or `DistroXV1Controller`. Registration is triggered by a `post` or `postInternal` call, and deletion is triggered by the various `delete*` method calls. I see in `StackV4Controller` that the methods that will trigger my code all have either a `@AccountId` or `@InitiatorUserCrn` annotation. This isn't true for the `DistroXV1Controller` calls. So I'm not sure exactly what changes I need to make.

Mihaly, you said:

> I'm not sure this is a good idea to change those instead of the calls which are really meant to be internal.

Which "calls which are really meant to be internal" are you referring to exactly? Do you mean the calls directly in the `FreeIPAEndpointManagementService`? Would it be more correct to move the `doAsInternalActor` to be inside those methods, instead of around the code making the method call?

Reply:
yes, you should add doAsInternal around `dnsV1Endpoint` eg:
```
ThreadBasedUserCrnProvider.doAsInternalActor(() -> dnsV1Endpoint.addDnsCnameRecord(request));